### PR TITLE
Uncertainty

### PR DESCRIPTION
**Pull request by AlexDuvalinho**
In this PR, we continue the work done on depfaenet and deup-depfaenet, integrating it into the active learning pipeline. 

**Aims:** 
(1) have a trained DepFAENet model fine-tuned on selected adsorbates, to predict the adsorption energies for catalyst-adsorbate pairs generated by GFlowNet
(2) have a deup-dataset which the intermediate representations of the above model, to train a Gaussian Process that measures uncertainty
(3) have a (v0) deup-depfaenet model that predicts uncertainty of the core DepFAENet model

**Main changes of this PR are:**
- Fix cluster issues ✅ 
- Update all associated configs and checkpoints relative to deup / depfaenet ✅ 
- Train a DepFAENet model, without concatenation, with tag0 and with dropout ✅  
- Obtain a deup dataset with above model ✅ 
- Train deup-depfaenet model that predicts the uncertainty ✅ 
- Ensure that it can be integrated to the active learning repo ✅ 
- Add a density estimator (q) for deup-faenet
- Add GP or DKL (replacing mc dropout ?) 

Information on best checkpoints are included in https://www.notion.so/Coding-Probabilistic-Surrogate-Model-328aece28af948f686fa91a2623d8635?pvs=4

----

**From the uncertainty-depfaenet branch**: 
- Import DepFAENet from disconnected_gnns Alvaro ✅
- Adapt DepFAENet (dropout, etc.) and pre-train it on full data ✅
- Re-train DepFAENet on selected adsorbates ✅
- Create deup-dataset using MC dropout depfaenet ✅
- Create Deup-DepFAENet ✅
- Train deup-depfaenet on deup-dataset ✅
- Active learning script to use deup-dataset ✅
- Fix imports issue ✅
- Allow FAENet Wrapper to keep track of hidden representations for active-learning part ✅

CAREFUL: only store graph-level rep in deup-dataset from now on. Need to change it if we want to perform MC dropout again (related to deup-faenet)

### CODE REVIEW

File: mila/sbatch.py [hunk 1]
```diff
@@ -41,7 +42,7 @@
     conda activate {env}
 fi
 {wandb_offline}
-srun --gpus-per-task=1 --output={output} {python_command}
```
Comment (vict0rsch) [inline mila/sbatch.py:44]: Anyone knows why this disappeared?
